appengine-java8/tasks/src/main/java/com/example/task/CreateTask.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.task;
+
+import com.google.cloud.tasks.v2beta2.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2beta2.CloudTasksClient;
+import com.google.cloud.tasks.v2beta2.HttpMethod;
+import com.google.cloud.tasks.v2beta2.QueueName;
+import com.google.cloud.tasks.v2beta2.Task;
+import com.google.common.base.Strings;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+
+import java.nio.charset.Charset;
+import java.time.Clock;
+import java.time.Instant;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class CreateTask {
+  private static String GGOGLE_CLOUD_PROJECT_KEY = "GOOGLE_CLOUD_PROJECT";
+
+  private static Option PROJECT_ID_OPTION = Option.builder("pid")
+      .longOpt("project-id")
+      .desc("The Google Cloud Project, if not set as GOOGLE_CLOUD_PROJECT env var.")
+      .hasArg()
+      .argName("project-id")
+      .type(String.class)
+      .build();
+
+  private static Option QUEUE_OPTION = Option.builder("q")
+      .required()
+      .longOpt("queue")
+      .desc("The Cloud Tasks queue.")
+      .hasArg()
+      .argName("queue")
+      .type(String.class)
+      .build();
+
+  private static Option LOCATION_OPTION = Option.builder("l")
+      .required()
+      .longOpt("location")
+      .desc("The region in which your queue is running.")
+      .hasArg()
+      .argName("location")
+      .type(String.class)
+      .build();
+
+  private static Option PAYLOAD_OPTION = Option.builder("p")
+      .longOpt("payload")
+      .desc("The payload string for the task.")
+      .hasArg()
+      .argName("payload")
+      .type(String.class)
+      .build();
+
+  private static Option IN_SECONDS_OPTION = Option.builder("s")
+      .longOpt("in-seconds")
+      .desc("Schedule time for the task to create.")
+      .hasArg()
+      .argName("in-seconds")
+      .type(int.class)
+      .build();
+
+  public static void main(String... args) throws Exception {
+    // [START cloud_tasks_appengine_create_task]
```

Comment:
Are you sure that you want the region tag to start here? It feels like most of the command line parsing isn't relevant enough that we want to link to it from elsewhere.

Reply:
I wasn't sure, but I also wanted to include enough context to understand what was happening. I don't know if these tags are used in the documentation, I was just copying them from the python sample. But I don't mind moving it.